providers/flagd/src/main/java/dev/openfeature/contrib/providers/flagd/resolver/rpc/RpcResolver.java

```diff
@@ -35,66 +37,96 @@
 import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
 import io.grpc.Status.Code;
 import io.grpc.StatusRuntimeException;
+import java.util.ArrayList;
+import java.util.List;
 import java.util.Map;
+import java.util.concurrent.LinkedBlockingQueue;
 import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicBoolean;
 import java.util.function.Consumer;
 import java.util.function.Function;
+import lombok.extern.slf4j.Slf4j;
 
 /**
  * Resolves flag values using https://buf.build/open-feature/flagd/docs/main:flagd.evaluation.v1.
  * Flags are evaluated remotely.
  */
 @SuppressWarnings("PMD.TooManyStaticImports")
 @SuppressFBWarnings(justification = "cache needs to be read and write by multiple objects")
-public final class GrpcResolver implements Resolver {
-
-    private final GrpcConnector<ServiceGrpc.ServiceStub, ServiceGrpc.ServiceBlockingStub> connector;
+@Slf4j
+public final class RpcResolver implements Resolver {
+    private static final int QUEUE_SIZE = 5;
+    private final AtomicBoolean shutdown = new AtomicBoolean(false);
+    private final ChannelConnector<ServiceStub, ServiceBlockingStub> connector;
     private final Cache cache;
     private final ResolveStrategy strategy;
     private final FlagdOptions options;
+    private final LinkedBlockingQueue<EventStreamResponseModel> incomingQueue;
+    private final Consumer<FlagdProviderEvent> onProviderEvent;
+    private final ServiceStub stub;
 
     /**
-     * Resolves flag values using https://buf.build/open-feature/flagd/docs/main:flagd.evaluation.v1.
+     * Resolves flag values using
+     * https://buf.build/open-feature/flagd/docs/main:flagd.evaluation.v1.
      * Flags are evaluated remotely.
      *
-     * @param options           flagd options
-     * @param cache             cache to use
+     * @param options         flagd options
+     * @param cache           cache to use
      * @param onProviderEvent lambda which handles changes in the connection/stream
      */
-    public GrpcResolver(
+    public RpcResolver(
             final FlagdOptions options, final Cache cache, final Consumer<FlagdProviderEvent> onProviderEvent) {
         this.cache = cache;
         this.strategy = ResolveFactory.getStrategy(options);
         this.options = options;
-        this.connector = new GrpcConnector<>(
-                options,
-                ServiceGrpc::newStub,
-                ServiceGrpc::newBlockingStub,
-                onProviderEvent,
-                stub -> stub.eventStream(
```

Comment:
This was moved into a private method so it can be called within our stream loop (same as `sync`).